earthfile2llb/converter.go

```diff
@@ -1140,7 +1140,20 @@ func (c *Converter) Expose(ctx context.Context, ports []string) error {
 	}
 	c.nonSaveCommand()
 	for _, port := range ports {
-		c.mts.Final.MainImage.Config.ExposedPorts[port] = struct{}{}
+		var exposedPort string
+		if strings.Contains(port, ":") {
+			port = strings.SplitN(port, ":", 2)[1]
+		}
+		if strings.Contains(port, "/") {
+			protocol := strings.SplitN(port, "/", 2)[1]
+			if protocol != "tcp" && protocol != "udp" {
+				return errors.New("Invalid protocol provided")
+			}
+			exposedPort = port
+		} else {
+			exposedPort = port + "/tcp"
```

Comment:
is this what docker does when no protocol is defined?

Reply:
yes if no protocol is present tcp is assigned by default